doc/basic_types.md

```diff
@@ -1,5 +1,360 @@
+JavaScript `null` or `undefined`, which are valid values.
+
+When C++ exceptions are disabled at compile time, a method with a `Value` 
+return type may return an empty value to indicate a pending exception. So when 
+not using C++ exceptions, callers should check whether the value is empty 
+before attempting to use it.
+
+#### Type
+```cpp
+napi_valuetype Type() const;
+```
+
+Gets the underlying `napi_valuetype` of the value.
+
+#### IsUndefined
+```cpp
+bool IsUndefined() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `undefined` or `false`
+otherwise.
+
+#### IsNull
+```cpp
+bool IsNull() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `null` or `false`
+otherwise.
+
+#### IsBoolean
+```cpp
+bool IsBoolean() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `true` or JavaScript 
+`false` or `false` if the value is not a Boolean value in JavaScript.
+
+#### IsNumber
+```cpp
+bool IsNumber() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Number` or `false`
+otherwise.
+
+#### IsString
+```cpp
+bool IsString() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `String` or `false`
+otherwise.
+
+#### IsSymbol
+```cpp
+bool IsSymbol() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Symbol` or `false`
+otherwise.
+
+#### IsArray
+```cpp
+bool IsArray() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Array` or `false`
+otherwise.
+
+#### IsArrayBuffer
+```cpp
+bool IsArrayBuffer() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `ArrayBuffer` or `false`
+otherwise.
+
+#### IsTypedArray
+```cpp
+bool IsTypedArray() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `TypedArray` or `false`
+otherwise.
+
+#### IsObject
+```cpp
+bool IsObject() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Object` or `false`
+otherwise.
+
+#### IsFunction
+```cpp
+bool IsFunction() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Function` or `false`
+otherwise.
+
+#### IsPromise
+```cpp
+bool IsPromise() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `Promise` or `false`
+otherwise.
+
+#### IsDataView
+```cpp
+bool IsDataView() const;
+```
+
+Returns `true` if the underlying value is a JavaScript `DataView` or `false`
+otherwise.
+
+#### IsBuffer
+```cpp
+bool IsBuffer() const;
+```
+
+Returns `true` if the underlying value is a Node `Buffer` or `false`
+otherwise.
+
+#### IsExternal
+```cpp
+bool IsExternal() const;
+```
+
+Returns `true` if the underlying value is a N-API external object or `false`
+otherwise.
+
+#### ToBoolean
+```cpp
+Boolean ToBoolean() const;
+```
+
+Converts a `Napi::Value` to a JavaScript boolean value. This is a wrapper 
+around `napi_coerce_to_boolean`. This will throw a JavaScript exception if the 
+coercion fails.
+
```

Comment:
Move this part on top and here something like this: Returns a JavaScript boolean value.